src/main/java/com/hw/bookstore/validation/UserRegistrationRequestDtoConstraintValidator.java

```diff
@@ -0,0 +1,39 @@
+package com.hw.bookstore.validation;
+
+import com.hw.bookstore.dto.request.UserRegistrationRequestDto;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Objects;
+
+public class UserRegistrationRequestDtoConstraintValidator

```

Comment:
This method validates only passwords so better name it like a PasswordValidator ;)